【OpenCL】add bilinear_interp op

chenjiaoAngel opened the pull request:
add bilinear_interp OP
add flatten2 OP
fix pad2d bug
fix resize bug

chenjiaoAngel committed:
Merge pull request #10 from PaddlePaddle/develop

update

chenjiaoAngel committed:
fix conv 2-pad to 4-pad

chenjiaoAngel committed:
fix compute conv shape

chenjiaoAngel committed:
fix pad, test=develop

chenjiaoAngel committed:
change conv_depthwise_3x3s1_fp.cc name to conv3x3s1p01_depthwise_fp32.cc to distinguish between conv3x3s1_depthwise_fp32.cc

chenjiaoAngel committed:
delete printf note in conv3x3s1, test=develop

chenjiaoAngel committed:
delete printf note, test=develop

chenjiaoAngel committed:
delete gem_sdot.h, test=develop

it is coped from __gemm_sdot_meta_.h

chenjiaoAngel committed:
update compute padding, test=develop

chenjiaoAngel committed:
fix padding size, must be 2 or 4. test=develop

chenjiaoAngel committed:
Merge pull request #13 from PaddlePaddle/develop

update code

chenjiaoAngel committed:
fix format in operators/conv_op.cc, test=develop

chenjiaoAngel committed:
change #if 0 to #if 1, test=develop

chenjiaoAngel committed:
put 2-pad to 4-pad in AttachImpl, test=develop

chenjiaoAngel committed:
fix clang-format error inn tests/math/connv_compute_test, test=develop

chenjiaoAngel committed:
fix x86 test result error, test=develop

chenjiaoAngel committed:
add asymmetric padding test case in liite/tests/math/conv_compute.cc, test=develop

chenjiaoAngel committed:
change paddings type to support dynamically modify, test=develop

chenjiaoAngel committed:
fix x86 build error in connv_compute_test, test=develop

chenjiaoAngel committed:
fix opencl build error, test=develop

chenjiaoAngel committed:
fix oopencl build error, test=develop

chenjiaoAngel committed:
fix  opencl/conv_compute build error, test=develop

chenjiaoAngel committed:
fix  opencl/conv_compute build error, test=develop

chenjiaoAngel committed:
fix format in kernels/opencl/conv_computte_ttest,test=develop

chenjiaoAngel committed:
Merge branch 'develop' into conv_pad

chenjiaoAngel committed:
Merge branch 'develop' into conv_pad

chenjiaoAngel committed:
fix build error, test=develop

fix build error in kernels/x86/conv_compute.h

chenjiaoAngel committed:
Merge pull request #20 from PaddlePaddle/develop

update

chenjiaoAngel committed:
fix ccompute shape error in ooperators/conv_op.h, test=develop

chenjiaoAngel committed:
update coode
Merge branch 'develop' of git://github.com/PaddlePaddle/Paddle-Lite into PaddlePaddle-develop

chenjiaoAngel committed:
get new code
Merge branch 'develop' of git://github.com/PaddlePaddle/Paddle-Lite into PaddlePaddle-develop

chenjiaoAngel committed:
add opencl concat kernell

chenjiaoAngel committed:
fix

chenjiaoAngel committed:
fix

chenjiaoAngel committed:
fix conflict

chenjiaoAngel committed:
fix conflic

chenjiaoAngel committed:
add model input test

chenjiaoAngel committed:
fix build error

chenjiaoAngel committed:
fix builld error

chenjiaoAngel committed:
add ut in opencl

chenjiaoAngel committed:
add ut

chenjiaoAngel committed:
fix kernel name

chenjiaoAngel committed:
fix kernel namedd

chenjiaoAngel committed:
fix ut

chenjiaoAngel committed:
fix utq

chenjiaoAngel committed:
add concat ut, test=develop

chenjiaoAngel committed:
Merge branch 'develop' into opencl

chenjiaoAngel committed:
fix axis compute, test=develop

chenjiaoAngel committed:
add other axis, test=develop

chenjiaoAngel committed:
fix ut. test=develop

chenjiaoAngel committed:
fix format

chenjiaoAngel committed:
delete extra files, test=develop

chenjiaoAngel committed:
fix origin, test=develop

chenjiaoAngel committed:
fix format. test=develop

chenjiaoAngel committed:
fix format2 test=develop

chenjiaoAngel committed:
Merge branch 'develop' into opencl

chenjiaoAngel committed:
Merge pull request #49 from PaddlePaddle/develop

update code

chenjiaoAngel committed:
fix conflict

chenjiaoAngel committed:
add leakyrelu and tanh compute

chenjiaoAngel committed:
add init val

chenjiaoAngel committed:
fix leakyrelu and tanh compute error, test=develop

chenjiaoAngel committed:
fix leakyrelu and tanh compute error, test=develop

chenjiaoAngel committed:
delete extra file, test=develop

chenjiaoAngel committed:
delete file

chenjiaoAngel committed:
fix format

chenjiaoAngel committed:
delete extra file, test=develop

chenjiaoAngel committed:
fix format， test=develop

chenjiaoAngel committed:
fix format. test=develop

chenjiaoAngel committed:
add grid sampler

chenjiaoAngel committed:
reset act

chenjiaoAngel committed:
fix conflict

chenjiaoAngel committed:
fix conflict and readme, test=develop

chenjiaoAngel committed:
fix ios run error, test=develop

chenjiaoAngel committed:
add exp/swish act op

chenjiaoAngel committed:
fix ios run error

chenjiaoAngel committed:
fix ios v8 run error

chenjiaoAngel committed:
fix grid_sampler compute error. test=develop

chenjiaoAngel committed:
fix exp act run error, test=develop

chenjiaoAngel committed:
fix format, test=develop

chenjiaoAngel committed:
Merge branch 'develop' into opencl

chenjiaoAngel committed:
fix format， test=develop

chenjiaoAngel committed:
fix redefine error

chenjiaoAngel committed:
fix format, test=develop

chenjiaoAngel committed:
add bilinear interp

chenjiaoAngel committed:
Merge pull request #63 from PaddlePaddle/develop

pull code

chenjiaoAngel committed:
fix formart

chenjiaoAngel committed:
pull code
Merge branch 'opencl' of https://github.com/chenjiaoAngel/Paddle-Lite into opencl

chenjiaoAngel committed:
fix build error

chenjiaoAngel committed:
fix conflict

chenjiaoAngel committed:
fix pad2d error

chenjiaoAngel committed:
fix format

chenjiaoAngel committed:
fixr

chenjiaoAngel committed:
fix

chenjiaoAngel committed:
fix format. test=develop

ysh329 commented on lite/kernels/opencl/CMakeLists.txt:
```diff
@@ -69,7 +69,9 @@ lite_cc_test(test_elementwise_add_image_opencl SRCS elementwise_add_image_comput
-
+lite_cc_test(test_bilinear_interp_image_opencl SRCS bilinear_interp_image_compute_test.cc
+	    DEPS bilinear_interp_opencl op_registry program context
+             ARGS --cl_path=${CMAKE_SOURCE_DIR}/lite/backends/opencl)
```
现在cl kernel已经打入so中了，不需要加`ARGS --cl_path=${CMAKE_SOURCE_DIR}/lite/backends/opencl)`

chenjiaoAngel replied:
hao

ysh329 commented on lite/kernels/opencl/bilinear_interp_image_compute.cc:
```diff
@@ -0,0 +1,176 @@
+  using param_t = operators::InterpolateParam;
+
+  std::string doc() const override {
+    return "BilinearInterp using cl::Image2D(ImageDefault/RGBA), kFP32";
```
```suggestion
    return "BilinearInterp using cl::Image2D(ImageDefault/RGBA), kFP16";
```

ysh329 commented on lite/kernels/opencl/bilinear_interp_image_compute.cc:
```diff
@@ -0,0 +1,176 @@
+
+    auto& context = ctx_->As<OpenCLContext>();
+    context.cl_context()->AddKernel(
+        kernel_func_name_, "image/bilinear_interp_kernel.cl", build_options_);
```
在这里加一行：
```cpp
VLOG(1) << "kernel_func_name_:" << kernel_func_name_;
```

ysh329 commented on lite/kernels/opencl/grid_sampler_image_compute.cc:
```diff
@@ -86,7 +86,8 @@ class GridSamplerImageCompute : public KernelLite<TARGET(kOpenCL),
                         DDim(std::vector<DDim::value_type>{
                             static_cast<int64_t>(out_image_shape["width"]),
                             static_cast<int64_t>(out_image_shape["height"])}));
-
```
在这个文件里的PrepareForRun里也加一行：
```cpp
VLOG(1) << "kernel_func_name_:" << kernel_func_name_;
```

chenjiaoAngel committed:
delete args and fix readme, test=devellop

chenjiaoAngel committed:
Merge branch 'opencl' of https://github.com/chenjiaoAngel/Paddle-Lite into opencl

pull code

chenjiaoAngel committed:
Merge branch 'develop' into opencl

chenjiaoAngel committed:
fix format, test=develop

chenjiaoAngel committed:
fix resize error

chenjiaoAngel committed:
pull
Merge branch 'opencl' of https://github.com/chenjiaoAngel/Paddle-Lite into opencl

chenjiaoAngel committed:
fix format. test=develop

chenjiaoAngel committed:
fix format, test=develop

chenjiaoAngel committed:
fix conflict, test=develop

ysh329 commented on lite/kernels/opencl/bilinear_interp_image_compute.cc:
```diff
@@ -0,0 +1,177 @@
+
+    auto out_image_shape = InitImageDimInfoWith(out_dims);
+    auto* x_img = x->data<half_t, cl::Image2D>();
+    VLOG(4) << "x_image: " << x_img;
```
注释掉这行哦

ysh329 replied:
打印没重载<< image，demo调用库时会报错

ysh329 commented on lite/kernels/opencl/bilinear_interp_image_compute.cc:
```diff
@@ -0,0 +1,177 @@
+
+    auto* out_img = out->mutable_data<half_t, cl::Image2D>(
+        out_image_shape["width"], out_image_shape["height"]);
+    VLOG(4) << "out_image: " << out_img;
```
注释掉这行哦

ysh329 commented on lite/kernels/opencl/CMakeLists.txt:
```diff
@@ -67,9 +67,10 @@ lite_cc_test(test_elementwise_add_image_opencl SRCS elementwise_add_image_comput
+             DEPS grid_sampler_opencl op_registry program context)
+             
+lite_cc_test(test_bilinear_interp_image_opencl SRCS bilinear_interp_image_compute_test.cc
+	    DEPS bilinear_interp_opencl op_registry program context)
```
下次记得对齐哦

ysh329 commented on lite/kernels/opencl/bilinear_interp_image_compute.cc:
```diff
@@ -0,0 +1,177 @@
+    auto& context = ctx_->As<OpenCLContext>();
+    context.cl_context()->AddKernel(
+        kernel_func_name_, "image/bilinear_interp_kernel.cl", build_options_);
+    VLOG(4) << "kernel_func_name_:" << kernel_func_name_;
```
```suggestion
    VLOG(1) << "kernel_func_name_:" << kernel_func_name_;
```

ysh329 replied:
VLOG(1)，会显示模型执行过程中所有PrepareForRun里的kernel

ysh329 commented on lite/kernels/opencl/grid_sampler_image_compute.cc:
```diff
@@ -44,7 +44,7 @@ class GridSamplerImageCompute : public KernelLite<TARGET(kOpenCL),
     context.cl_context()->AddKernel(
         kernel_func_name_, "image/grid_sampler_kernel.cl", build_options_);
-    VLOG(1) << "kernel_func_name_:" << kernel_func_name_;
+    VLOG(4) << "kernel_func_name_:" << kernel_func_name_;
```
```suggestion
    VLOG(1) << "kernel_func_name_:" << kernel_func_name_;
```
级别改成1

chenjiaoAngel committed:
fix format

chenjiaoAngel committed:
note log(4) << x_image. test=develop

ysh329 reviewed (dismissed):
LGTM

chenjiaoAngel committed:
note x_image printf. test=develop
